certif/tests/integration/components/login-session-supervisor-form_test.js

```diff
@@ -11,11 +11,54 @@ module('Integration | Component | login-session-supervisor-form', function(hooks
 
   test('it should render supervisor login form', async function(assert) {
     // when
-    const screen = await renderScreen(hbs`<LoginSessionSupervisorForm />`);
+    this.onFormSubmit = sinon.stub();
+    const screen = await renderScreen(hbs`<LoginSessionSupervisorForm @onFormSubmit={{this.onFormSubmit}} />`);
 
     // then
     assert.dom(screen.getByLabelText('Numéro de la session')).exists();
     assert.dom(screen.getByLabelText('Mot de passe de la session')).exists();
     assert.dom(screen.getByText('Surveiller la session')).exists();
   });
+
+  module('On click on supervise button', function() {
+    test('it should display an error message when the session id is empty', async function(assert) {
+      // given
+      this.onFormSubmit = sinon.stub();
+      const screen = await renderScreen(hbs`<LoginSessionSupervisorForm @onFormSubmit={{this.onFormSubmit}} />`);
+      await fillIn(screen.getByLabelText('Mot de passe de la session'), '12345');
+
+      // when
+      await click(screen.getByText('Surveiller la session'));
+
+      // then
+      assert.contains('Les champs "Numéro de la session" et "Mot de passe de session" sont obligatoires.');
+    });
+
+    test('it should display an error message when the supervisor password is empty', async function(assert) {
+      // given
+      this.onFormSubmit = sinon.stub();
+      const screen = await renderScreen(hbs`<LoginSessionSupervisorForm @onFormSubmit={{this.onFormSubmit}} />`);
+      await fillIn(screen.getByLabelText('Numéro de la session'), '12345');
+
+      // when
+      await click(screen.getByText('Surveiller la session'));
+
+      // then
+      assert.contains('Les champs "Numéro de la session" et "Mot de passe de session" sont obligatoires.');
+    });
+
+    test('it should call onFormSubmit when all the fields are filled', async function(assert) {
+      // given
+      this.onFormSubmit = sinon.stub();
+      const screen = await renderScreen(hbs`<LoginSessionSupervisorForm @onFormSubmit={{this.onFormSubmit}} />`);
+      await fillIn(screen.getByLabelText('Numéro de la session'), '12345');
+      await fillIn(screen.getByLabelText('Mot de passe de la session'), '12345');
+
+      // when
+      await click(screen.getByText('Surveiller la session'));
+
+      // then
+      assert.ok(this.onFormSubmit.calledWith({ sessionId: '12345', supervisorPassword: '12345' }));
```

Comment:
```suggestion
sinon.assert.calledWith(this.onFormSubmit, { sessionId: '12345', supervisorPassword: '12345' }));
assert(true)
```
cons: le assert true
pros:  le message d'erreur explicite si le calledWith ne fonctionne pas

Reply:
Done